src/core/qgsvectorlayer.cpp

```diff
@@ -4335,6 +4341,27 @@ QString QgsVectorLayer::getStyleFromDatabase( const QString& styleId, QString &m
   return getStyleByIdMethod( mDataSource, styleId, msgError );
 }
 
+bool QgsVectorLayer::deleteStyleFromDatabase( const QString& styleId, QString &msgError )
+{
+  QgsProviderRegistry * pReg = QgsProviderRegistry::instance();
+  QLibrary *myLib = pReg->providerLibrary( mProviderKey );
```

Comment:
Sorry @nyalldawson. I didn't understand your comment. Where should I use `QScopedPointer`?